pkg/webhooks/machine_webhook.go

```diff
@@ -724,13 +726,9 @@ func validateAWS(m *machinev1beta1.Machine, config *admissionConfig) (bool, []st
 	}
 
 	if providerSpec.UserDataSecret == nil {
-		errs = append(
-			errs,
-			field.Required(
-				field.NewPath("providerSpec", "userDataSecret"),
-				"expected providerSpec.userDataSecret to be populated",
-			),
-		)
+		errs = append(errs, field.Required(field.NewPath("providerSpec", "userDataSecret"), "expected providerSpec.userDataSecret to be populated"))
+	} else if providerSpec.UserDataSecret.Name == "" {
+		errs = append(errs, field.Required(field.NewPath("providerSpec", "userDataSecret", "name"), "name must be provided"))
```

Comment:
Units are failing as there is a discrepancy in the expectedError
```suggestion
		errs = append(errs, field.Required(field.NewPath("providerSpec", "userDataSecret", "name"), "expected providerSpec.userDataSecret.name to be provided"))
```